### PR TITLE
feat: add atlas step to the build - FC-0012

### PR DESCRIPTION
**Pull request by shadinaif**
Adding the pull behind a feature flag until it's fully implemented

**IMPORTANT:** This PR needs these to be merged first
- https://github.com/openedx/ecommerce/pull/4051 
- https://github.com/openedx/ecommerce/pull/4108

Background
---

This contribution is part of the [FC-0012 project](https://openedx.atlassian.net/l/cp/XGS0iCcQ) which is sparked by the [Translation Infrastructure update OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html#specification).

### CODE REVIEW

**Comment by shadinaif** [comment]
Please review @omarithawi, @regisb

File: tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile [hunk 1]
```diff
@@ -79,6 +79,13 @@ RUN cd /openedx/requirements/ \
 {% for extra_requirement in ECOMMERCE_EXTRA_PIP_REQUIREMENTS %}RUN {% if is_buildkit_enabled() %}--mount=type=cache,target=/openedx/.cache/pip,sharing=shared,uid=${APP_USER_ID} {% endif %}pip install '{{ extra_requirement }}'
 {% endfor %}
 
+{% if ECOMMERCE_ATLAS_PULL %}
+# Pull translations. Support the OEP-58 proposal behind a feature flag until it's fully implemented.
+RUN find ecommerce/conf/locale -mindepth 1 -maxdepth 1 -type d -exec rm -r {} \;
```
Comment (OmarIthawi) [inline tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile]: This line is discouraged from being used in the _Tutor plugins_. See Regis comment: https://github.com/overhangio/tutor-discovery/pull/41#discussion_r1316002004 .

```suggestion
```

**Review by OmarIthawi — COMMENTED** [review]
One change and this should look good. Thanks @shadinaif!

File: tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile [hunk 2]
```diff
@@ -5,7 +5,7 @@ FROM docker.io/ubuntu:20.04 as minimal
 RUN {% if is_buildkit_enabled() %}--mount=type=cache,target=/var/cache/apt,sharing=locked \
```
Comment (OmarIthawi) [inline tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile]: @shadinaif I've found that the build gets stuck asking for user input which won't work during `docker build`, unless the `DEBIAN_FRONTEND=noninteractive` is set.

See https://techoverflow.net/2019/05/18/how-to-fix-configuring-tzdata-interactive-input-when-building-docker-images/

I've noticed that this variable is set as well in Tutor Discovery:

 - https://github.com/overhangio/tutor-discovery/blame/af757b3e4c31e0f41c9c8014b1094dd94e2decbc/tutordiscovery/templates/discovery/build/discovery/Dockerfile#L5

Comment (shadinaif) [reply tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile]: Done

File: tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile (hunk 1 above)
Comment (shadinaif) [reply tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile]: done

**Comment by shadinaif** [comment]
Done @OmarIthawi . Also, this PR needs https://github.com/openedx/ecommerce/pull/4051 to be merged first. And that's for one reason: atlas is added to requirements in ecommerce PR

**Review by OmarIthawi — COMMENTED** [review]
Thanks @shadinaif! Looks good to me.

I will test again after the ecommerce pull request is merged.

**Comment by brian-smith-tcril** [comment]
@OmarIthawi https://github.com/openedx/ecommerce/pull/4051 has been merged.

@shadinaif it looks like this one needs a rebase

**Comment by OmarIthawi** [comment]
Thanks @brian-smith-tcril for taking care of the ecommerce PR! Yup, Quince has been tagged. So this one will land post-quince.

We'll rebase, test and let you know when it's ready.

**Comment by shadinaif** [comment]
rebased, thank you @brian-smith-tcril

File: tutorecommerce/plugin.py [hunk 3]
```diff
@@ -31,6 +31,7 @@
         "OAUTH2_KEY_SSO": "ecommerce-sso",
         "OAUTH2_KEY_SSO_DEV": "ecommerce-sso-dev",
         "WORKER_JWT_ISSUER": "ecommerce-worker",  # TODO do we need to keep this?
+        "ATLAS_PULL": False,
```
Comment (OmarIthawi) [inline tutorecommerce/plugin.py]: Please remove this variable. No feature flag is required anymore.

File: tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile [hunk 4]
```diff
@@ -79,6 +80,12 @@ RUN cd /openedx/requirements/ \
 {% for extra_requirement in ECOMMERCE_EXTRA_PIP_REQUIREMENTS %}RUN --mount=type=cache,target=/openedx/.cache/pip,sharing=shared,uid=${APP_USER_ID} pip install '{{ extra_requirement }}'
 {% endfor %}
 
+{% if ECOMMERCE_ATLAS_PULL %}
+# Pull translations. Support the OEP-58 proposal behind a feature flag until it's fully implemented.
+RUN atlas pull {{ patch("atlas-extra-args") }} translations/ecommerce/ecommerce/conf/locale:ecommerce/conf/locale
```
Comment (OmarIthawi) [inline tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile]: Please use the new format as seen in https://github.com/overhangio/tutor-credentials/pull/29/files#diff-1d0fcae46139cd49acb181fdd0a46cae537e257b78120b14bdc283b2db30054aR128

**Review by OmarIthawi — COMMENTED** [review]
@shadinaif please refactor in favor of the new integration:

 - https://github.com/overhangio/tutor-credentials/pull/29
 - https://github.com/overhangio/tutor/pull/964

File: tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile [hunk 5]
```diff
@@ -79,6 +80,9 @@ RUN cd /openedx/requirements/ \
 {% for extra_requirement in ECOMMERCE_EXTRA_PIP_REQUIREMENTS %}RUN --mount=type=cache,target=/openedx/.cache/pip,sharing=shared,uid=${APP_USER_ID} pip install '{{ extra_requirement }}'
 {% endfor %}
 
+RUN atlas pull --repository="{{ ATLAS_REPOSITORY }}" --branch="{{ ATLAS_REVISION }}" {{ ATLAS_OPTIONS }} translations/ecommerce/ecommerce/conf/locale:ecommerce/conf/locale
```
Comment (OmarIthawi) [inline tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile]: @shadinaif There's a very new addition to atlas:

 - https://github.com/openedx/openedx-atlas/pull/49

Would you like to implement it in this pull request or create a new one after this one is merged?

```suggestion
RUN atlas pull --repository="{{ ATLAS_REPOSITORY }}" --revision="{{ ATLAS_REVISION }}" {{ ATLAS_OPTIONS }} translations/ecommerce/ecommerce/conf/locale:ecommerce/conf/locale
```

cc: @Faraz32123 @brian-smith-tcril

**Review by OmarIthawi — APPROVED** [review]
LGTM. I think this is ready to get merged unless we want to use the new `--revision` argument.

File: tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile (hunk 5 above)
Comment (brian-smith-tcril) [reply tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile]: @shadinaif could you please accept that suggestion?

Comment (OmarIthawi) [reply tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile]: Meanwhile, I've upgraded the ecommerce PR:

 - https://github.com/openedx/ecommerce/pull/4108

We need to merge that one first.

Comment (shadinaif) [reply tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile]: updated and rebased, waiting for [4108](https://github.com/openedx/ecommerce/pull/4108) to be merged

**Comment by OmarIthawi** [comment]
@Faraz32123 This is now ready for merge.

Thanks @shadinaif for the work.

**Comment by Faraz32123** [comment]
@shadinaif please add changelog entry.

**Comment by shadinaif** [comment]
Change log added @Faraz32123 , thank you!

**Review by Faraz32123 — APPROVED** [review]
LGTM!

File: changelog.d/20240125_122858_shadinaif_FC_0012_OEP_58.md [hunk 6]
```diff
@@ -0,0 +1 @@
+💥[Feature] Add support to `atlas pull` - FC-0012 project, OEP-58. (by @shadinaif) -->
```
Comment (OmarIthawi) [inline changelog.d/20240125_122858_shadinaif_FC_0012_OEP_58.md:1]: @shadinaif I found few typos after merging this pull request. Please fix them if needed.

```suggestion
- [Feature] Add support to `atlas pull` - FC-0012 project, OEP-58. (by @shadinaif)
```

cc: @Faraz32123

Comment (shadinaif) [reply changelog.d/20240125_122858_shadinaif_FC_0012_OEP_58.md:1]: thank @OmarIthawi , fixed here https://github.com/overhangio/tutor-ecommerce/pull/66

**Pull request merged**